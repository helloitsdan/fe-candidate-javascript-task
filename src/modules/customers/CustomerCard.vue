<template>
  <div class="card">
    <h3 class="is-marginless">{{ name | formatName }}</h3>

    <p>{{ email }}</p>

    <p>
      <strong>Registered on</strong>
      {{ registered | moment("dddd, MMMM Do YYYY") }}
    </p>

    <p v-if="!isActive">
      This customer is
      <strong>inactive.</strong>
    </p>

    <p class="is-marginless">
      <router-link
        :to="{ name: 'CustomerDetails', params: { customerId: uuid } }"
      >
        View details
      </router-link>
    </p>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "CustomerCard",
  props: {
    uuid: String,
    isActive: Boolean,
    name: Object,
    email: String,
    registered: String
  }
});
</script>
