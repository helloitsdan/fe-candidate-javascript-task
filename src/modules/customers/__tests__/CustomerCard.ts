import { shallowMount } from "@vue/test-utils";

import CustomerCard from "../CustomerCard.vue";

describe("CustomerCard", () => {
  const fakeCustomer = {
    uuid: "1234",
    isActive: true,
    name: {
      first: "Test",
      last: "Customer"
    },
    email: "test.user@pretend.biz",
    registered: "2016-07-13T02:57:55.235Z"
  };

  it("renders", () => {
    const wrapper = shallowMount(CustomerCard, {
      propsData: fakeCustomer,
      stubs: ["router-link"]
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
