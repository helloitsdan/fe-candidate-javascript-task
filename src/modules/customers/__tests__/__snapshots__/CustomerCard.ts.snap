// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerCard renders 1`] = `
<div
  class="card"
>
  <h3
    class="is-marginless"
  >
    Test Customer
  </h3>
   
  <p>
    test.user@pretend.biz
  </p>
   
  <p>
    <strong>
      Registered on
    </strong>
    
    Wednesday, July 13th 2016
  
  </p>
   
  <!---->
   
  <p
    class="is-marginless"
  >
    <router-link-stub
      to="[object Object]"
    >
      
      View details
    
    </router-link-stub>
  </p>
</div>
`;
